test/functional/p2p_orphan_handling.py

```diff
@@ -591,6 +591,30 @@ def test_orphan_txid_inv(self):
         assert_equal(node.getmempoolentry(tx_child["txid"])["wtxid"], tx_child["wtxid"])
         self.wait_until(lambda: len(node.getorphantxs()) == 0)
 
+    @cleanup
+    def test_immediately_erased_orphan(self):
+        self.log.info("Check that we fetch parents of an orphan, even if the orphan is not persisted in orphanage")
+
+        # Causes immediate eviction from the orphanage which normally happens randomly with
+        # 1/(DEFAULT_MAX_ORPHAN_TRANSACTIONS + 1) chance when orphanage is full
+        self.restart_node(0, extra_args=["-maxorphantx=0"])
+
+        tx_parent = self.wallet.create_self_transfer()
+        tx_child = self.wallet.create_self_transfer(utxo_to_spend=tx_parent["new_utxo"])
+
+        peer = self.nodes[0].add_p2p_connection(P2PInterface())
+
+        peer.send_and_ping(msg_tx(tx_child["tx"]))
+        assert_equal(len(self.nodes[0].getorphantxs()), 0)
```

Comment:
Would probably move this to the very end, given what we know about `getorphantxs` taking a while to update